src/coreclr/src/vm/comdelegate.cpp

```diff
@@ -1588,6 +1596,10 @@ FCIMPL3(void, COMDelegate::DelegateConstruct, Object* refThisUNSAFE, Object* tar
     // try to catch the easy garbage.
     _ASSERTE(isMemoryReadable(method, 1));
 
+#if defined(HOST_OSX) && defined(HOST_ARM64)
```

Comment:
Dtto. (If you are just hacking around to make something work and all this is expected to be reworked, let me know and I will stop commenting on these.)

Reply:
This initial implementation is a bit of a hack.  Primary goal was to unblock JIT ABI work.

@janvorli intends to create a platform generic solution which might look substantially different.

I was trying to put most of these as high in the call stack as possible to avoid making tons of little calls to the OS to toggle these pages.  However, I was forced to move some of them lower as lldb was unstable and would crash when getting a call stack.

As I am looking at this I wonder about exceptions, so I think we may need a holder mechanism so we can guarantee the thread state is properly restored. 

I am thinking it would be good to have a more structured code manager, where writes to code are queued and centrally written and flushed etc.  This seems very scattered....